### PR TITLE
Issue #358 Adding manual instructions to use SSHFS host folder mounts

### PR DESCRIPTION
**Pull request by hferentschik**

### CODE REVIEW

File: docs/managing-minishift.md [hunk 1]
```diff
@@ -181,8 +183,20 @@ These are not configurable at the moment and are different for each driver and t
 <a name="mounting-custom-shared-folders"></a>
 ### Mounting custom shared folders
 
-The Boot2Docker and the CentOS image include `cifs-utils`, which allows you to mount CIFS-based shared
-folders inside the VM. For example, on Windows 10 the `C:\Users` folder is shared and only needs locally
+The Minishift [Boot2Docker](https://github.com/minishift/minishift-b2d-iso) as well as the
+Minishift [CentOS](https://github.com/minishift/minishift-centos-iso) image include `cifs-utils` and
```
Comment (LalatenduMohanty) [inline docs/managing-minishift.md]: s/image include/ISO include/

File: docs/managing-minishift.md [hunk 2]
```diff
@@ -181,8 +183,20 @@ These are not configurable at the moment and are different for each driver and t
...
+
+**Note:** When using the Boot2Docker image in conjunction with the VirtualBox driver, Virtual Box
+guest additions are automatically enabled and occupy the _/Users_ mountpoint used in the following
+examples. In this case you will need to use a different mountpoint.
+
+<a name="cifs"></a>
+#### CIFS
+
+An example for CIFS-based sharing is the mount of `C:\Users` on a Windows host into the Minishift VM.
+On Windows 10 the `C:\Users` folder is shared per default and only needs locally
```
Comment (LalatenduMohanty) [inline docs/managing-minishift.md]: Not sure if should be `locally authenticated users` or `local authenticated users`

Comment (hferentschik) [inline docs/managing-minishift.md]: I am not 100% sure either. Given that this has already been reviewed once, I guess locally should be ok

File: docs/managing-minishift.md (hunk 1 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: Ok

**Comment by hferentschik** [comment]
Rebased and applied review suggestion

File: docs/managing-minishift.md [hunk 3]
```diff
@@ -181,8 +183,20 @@ These are not configurable at the moment and are different for each driver and t
...
 
-The Boot2Docker and the CentOS image include `cifs-utils`, which allows you to mount CIFS-based shared
-folders inside the VM. For example, on Windows 10 the `C:\Users` folder is shared and only needs locally
+The Minishift [Boot2Docker](https://github.com/minishift/minishift-b2d-iso) as well as the
+Minishift [CentOS](https://github.com/minishift/minishift-centos-iso) ISO include `cifs-utils` and
+`fuse-sshfs`. This allows you to mount host folders using [CIFS](https://en.wikipedia.org/wiki/Server_Message_Block)
+or [SSHFS](https://en.wikipedia.org/wiki/SSHFS).
+
+**Note:** When using the Boot2Docker image in conjunction with the VirtualBox driver, Virtual Box
+guest additions are automatically enabled and occupy the _/Users_ mountpoint used in the following
```
Comment (Preeticp) [inline docs/managing-minishift.md]: When using the Boot2Docker image in conjunction with the VirtualBox driver> When you use the Boot2Docker image along with the VirtualBox driver,...

File: docs/managing-minishift.md [hunk 4]
```diff
@@ -181,8 +183,20 @@ These are not configurable at the moment and are different for each driver and t
...
+examples. In this case you will need to use a different mountpoint.
+
```
Comment (Preeticp) [inline docs/managing-minishift.md]: Virtual Box guest additions are automatically enabled and occupy the _/Users_ mountpoint used in the following examples.> ...as shown in the following examples?

File: docs/managing-minishift.md [hunk 5]
```diff
@@ -181,8 +183,20 @@ These are not configurable at the moment and are different for each driver and t
 <a name="mounting-custom-shared-folders"></a>
 ### Mounting custom shared folders
 
-The Boot2Docker and the CentOS image include `cifs-utils`, which allows you to mount CIFS-based shared
-folders inside the VM. For example, on Windows 10 the `C:\Users` folder is shared and only needs locally
+The Minishift [Boot2Docker](https://github.com/minishift/minishift-b2d-iso) as well as the
+Minishift [CentOS](https://github.com/minishift/minishift-centos-iso) ISO include `cifs-utils` and
```
Comment (Preeticp) [inline docs/managing-minishift.md]: ISO>ISOs

File: docs/managing-minishift.md [hunk 6]
```diff
@@ -217,6 +231,36 @@ authenticated users. The following procedure describes how to mount this folder.
 - If your Windows account is linked to a Microsoft account, you must use the full Microsoft account email address to
   authenticate, for example `jpillow@amigas.us`. This ensures that the domain value that contains the computer name is provided.
 
+
+<a name="sshfs"></a>
+#### SSHFS
+
+You can use SSHFS based host folder mounts, when you have a SSH daemon running on your host. For Linux
```
Comment (Preeticp) [inline docs/managing-minishift.md]: host folder mounts,> no comma required

File: docs/managing-minishift.md [hunk 7]
```diff
@@ -217,6 +231,36 @@ authenticated users. The following procedure describes how to mount this folder.
...
+and OS X this prerequisite is most of the time already met. Most Linux distributions have a SSH daemon
```
Comment (Preeticp) [inline docs/managing-minishift.md]: Mostly, this prerequisite is met by default on Linux and OS X.

File: docs/managing-minishift.md [hunk 8]
```diff
@@ -217,6 +231,36 @@ authenticated users. The following procedure describes how to mount this folder.
...
+installed per default. If not, follow the instructions for your specific distribution to install one.
+OS X also has a built-in SSH server. To use it, make sure _Remote Login_ is enabled under
+_System Preferences > Sharing_. On Windows you should consider using a [CIFS based](#cifs)
```
Comment (Preeticp) [inline docs/managing-minishift.md]: On Windows sentence should be on the next line, different OS as well as different suggestion for the OS.

File: docs/managing-minishift.md [hunk 9]
```diff
@@ -217,6 +231,36 @@ authenticated users. The following procedure describes how to mount this folder.
...
+mount first, but if you want to try SSHFS, you can install [OpenSSH for Windows](https://winscp.net/eng/docs/guide_windows_openssh_server).
+
+1. Similar to the [CIFS]((#cifs)) example, you need the IP address from which your host is reachable
```
Comment (Preeticp) [inline docs/managing-minishift.md]: double brackets on #cifs

File: docs/managing-minishift.md (hunk 3 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: ok, thanks

File: docs/managing-minishift.md (hunk 4 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: Ok

File: docs/managing-minishift.md (hunk 5 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: Really?!

File: docs/managing-minishift.md (hunk 6 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: ok

File: docs/managing-minishift.md (hunk 7 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: 👍

File: docs/managing-minishift.md (hunk 8 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: +1

File: docs/managing-minishift.md (hunk 9 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: thanks

File: docs/managing-minishift.md [hunk 10]
```diff
@@ -217,6 +231,36 @@ authenticated users. The following procedure describes how to mount this folder.
...
+from within the VM. Run `ifconfig` (or `Get-NetIPAddress` on Windows) to determine the local IP
```
Comment (Preeticp) [inline docs/managing-minishift.md]: you need the local IP address so that your host is reachable from within the VM?

Comment (hferentschik) [inline docs/managing-minishift.md]: > you need the local IP address so that your host is reachable from within the VM?

-1, this is semantically wrong.

**Comment by hferentschik** [comment]
@Preeticp thanks for the review. I applied most of you suggestions.

File: docs/managing-minishift.md (hunk 5 above)
Comment (LalatenduMohanty) [inline docs/managing-minishift.md]: I think it should be `ISO`.

Comment (Preeticp) [inline docs/managing-minishift.md]: We are speaking of 2 ISOs here right? the B2D and the Centos one?

File: docs/managing-minishift.md [hunk 11]
```diff
@@ -181,8 +183,20 @@ These are not configurable at the moment and are different for each driver and t
...
 ### Mounting custom shared folders
 
-The Boot2Docker and the CentOS image include `cifs-utils`, which allows you to mount CIFS-based shared
-folders inside the VM. For example, on Windows 10 the `C:\Users` folder is shared and only needs locally
+The Minishift [Boot2Docker](https://github.com/minishift/minishift-b2d-iso) as well as the
+Minishift [CentOS](https://github.com/minishift/minishift-centos-iso) ISO include `cifs-utils` and
+`fuse-sshfs`. This allows you to mount host folders using [CIFS](https://en.wikipedia.org/wiki/Server_Message_Block)
+or [SSHFS](https://en.wikipedia.org/wiki/SSHFS).
+
+**Note:** When you use the Boot2Docker ISO along with the VirtualBox driver, Virtual Box
```
Comment (budhrg) [inline docs/managing-minishift.md]: `Virtual Box` should be `Virtualbox`.

File: docs/managing-minishift.md [hunk 12]
```diff
@@ -181,8 +183,20 @@ These are not configurable at the moment and are different for each driver and t
 <a name="mounting-custom-shared-folders"></a>
 ### Mounting custom shared folders
 
-The Boot2Docker and the CentOS image include `cifs-utils`, which allows you to mount CIFS-based shared
```
Comment (budhrg) [inline docs/managing-minishift.md]: There is a broken link in the section `mounted-host-folders` _Note_ section. Could you also fix it as part of this PR @hferentschik ?

File: docs/managing-minishift.md (hunk 11 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: Fixed

File: docs/managing-minishift.md (hunk 12 above)
Comment (hferentschik) [inline docs/managing-minishift.md]: fixed

File: docs/managing-minishift.md (hunk 5 above)
Comment (thatdocslady) [inline docs/managing-minishift.md]: ISOs is not an official term, I would qualify them separately (as in the x ISO and the y ISO) to ease the grammar thing :)

**Pull request merged**